src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -11449,22 +11497,25 @@ private NullableFlowState GetValue(int id, int index)
                     Debug.Assert(Id > id);
                     return _container!.Value.GetValue(id, index);
                 }
-                else
+                return GetValue(index);
+            }
+
+            private NullableFlowState GetValue(int index)
+            {
+                if (!this.Reachable)
                 {
-                    if (index < Capacity && this.Reachable)
-                    {
-                        index *= 2;
-                        var result = (_state[index + 1], _state[index]) switch
-                        {
-                            (false, false) => NullableFlowState.NotNull,
-                            (false, true) => NullableFlowState.MaybeNull,
-                            (true, false) => throw ExceptionUtilities.UnexpectedValue(index),
-                            (true, true) => NullableFlowState.MaybeDefault
-                        };
-                        return result;
-                    }
                     return NullableFlowState.NotNull;
                 }
+                Debug.Assert(index < Capacity);
+                index *= 2;
+                var result = (_state[index], _state[index + 1]) switch
+                {
+                    (false, false) => throw ExceptionUtilities.Unreachable(),
```

Comment:
The reason we're throwing `Unreachable` here is because we *never* expect this bit pattern to be present in regions of the bit vector which are within capacity, right?

Reply:
Right. We are deliberately avoiding any reliance on default values. Since we only need 3 possible combinations, the false-false is outlawed ;-)